### PR TITLE
save old legend and delete it to avoid remigration

### LINKED ISSUE #10410: Legends are migrated if no legends set always

### PR DESCRIPTION
**Pull request by gfiorav**
Fixes #10410 

Please CR @javitonino

### CODE REVIEW

File: app/controllers/carto/api/vizjson3_presenter.rb [hunk 1]
```diff
@@ -335,7 +335,11 @@ def as_data
       end
 
       def migrate_legends(old_legend)
+        @layer.options[:backup_legend] = old_legend
         Carto::LegendMigrator.new(@layer.id, old_legend).build.save
+        @layer.options[:legend] = nil
```
Comment (javitonino) [inline app/controllers/carto/api/vizjson3_presenter.rb]: `@layer.options.remove(:legend)` ?

**Comment by gfiorav** [comment]
The thing is that there might already be a guy in the builder who's legends need to be migrated because we didn't back in the day or we need to do it again. I hadn't thought on version to be honest. What do you think?

**Comment by javitonino** [comment]
Uhm, you are right that this offers some advantages. :+1:. Then, maybe instead of renaming `legend` -> `backup_legend`, do something like `options[:legend][:migrated] = true`, so it's easier to go back to the editor (just in case).

**Comment by gfiorav** [comment]
Have a quick second look please, @javitonino and I'll merge!

**Comment by javitonino** [comment]
LGTM

**Comment by javitonino** [comment]
Quick thought: Maybe it is worth it to mark legends as migrated as soon as we see something in the new legend model. My scenario is that current users that have migrated a map (previous to this patch) and then delete all legends, will get a legend migration (just once) which may be annoying. But it is only one time, so...

**Comment by gfiorav** [comment]
Yup, totally. Changing that, thanks!

**Comment by gfiorav** [comment]
How about now?

**Comment by gfiorav** [comment]
@javitonino Ok I hope it's good now!

File: app/controllers/carto/api/vizjson3_presenter.rb [hunk 2]
```diff
@@ -334,8 +335,14 @@ def as_data
...
-        Carto::LegendMigrator.new(@layer.id, old_legend).build.save
+      def migrate_old_legend
+        Carto::LegendMigrator.new(@layer.id, @layer.legend).build.save
+        mark_old_legend_migrated
+      end
+
+      def mark_old_legend_migrated
+        @layer.options[:legend][:migrated] = true
+        @layer.save
         @layer.legends.reload
```
Comment (javitonino) [inline app/controllers/carto/api/vizjson3_presenter.rb]: I would move `@layer.legends.reload` to `migrate_old_legend`. Not a big deal, but seems to make more sense.

**Comment by gfiorav** [comment]
Thanks, one last look.

**Comment by javitonino** [comment]
:+1:

**Pull request merged**